doc/arch/adr-002-rest-api.md

```diff
@@ -0,0 +1,77 @@
+| `/auth/login/`  | GET    | -         | Allow Any      | 200 OK,<br />401 Not Authorized                       | Returns loging JSON if current session valid |
+| `/auth/login/`  | POST   | `{ ... }` | Allow Any      | 200 OK,<br />400 Bad Request,<br />401 Not Authorized | Used for starting new valid session          |
+| `/auth/logout/` | POST   | -         | Logged In User | 200 OK                                                | Used for logging out                         |
+
+Also present are `/auth/password/reset/` and `/auth/password/reset/confirm/` endpoints that come from `dj_rest_auth`.
+
+### Boundaries
+
+| Path                                       | Method | Body                     | Authentication | Responses                                                  | Notes                                                                                                              |
+| ------------------------------------------ | ------ | ------------------------ | -------------- | ---------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------ |
+| `/boundaries/`                             | GET    | -                        | Logged In User | 200 OK                                                     | Returns list of all boundaries the user has access to                                                              |
+| `/boundaries/?utilities=1,3,5,8`           | GET    | -                        | Logged In User | 200 OK,<br />401 Not Authorized                            | Returns list of boundaries in the given utilities. If the user does not have access to that utility, returns a 401 |
+| `/boundaries/`                             | POST   | `{ ... }`                | Contributor    | 201 Created,<br />400 Bad Request,<br />401 Not Authorized | Creates a new boundary if the payload is correct and the user is authorized                                        |
+| `/boundaries/{id}/`                        | GET    | -                        | Logged In User | 200 OK,<br />404 Not Found                                 | Returns boundary details if the user has access to it, else 404s. Includes all details about the boundary.         |
+| `/boundaries/{id}/shape/`                  | PUT    | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Updates the latest submission's shape                                                                              |
+| `/boundaries/{id}/reference-images/`       | POST   | `{ ... }`                | Contributor    | 200 OK,<br />404 Not Found                                 | Adds a new reference image to the boundary                                                                         |
```

Comment:
I think we need another endpoint for updating specific reference images (visibility, distortion).

Reply:
[4e79c16](https://github.com/azavea/iow-boundary-tool/pull/115/commits/4e79c1642faea9e0bda5fe6ad924e227850b67e7)